python/lsst/ctrl/mpexec/singleQuantumExecutor.py

```diff
@@ -164,26 +163,19 @@ def __init__(
                     collectionTypes=CollectionType.RUN,
                 )
 
-    def execute(self, task_node: TaskDef | TaskNode, /, quantum: Quantum) -> Quantum:
+    def execute(
+        self, task_node: TaskNode | TaskDef, /, quantum: Quantum
+    ) -> tuple[Quantum, QuantumReport | None]:
         # Docstring inherited from QuantumExecutor.execute
         assert quantum.dataId is not None, "Quantum DataId cannot be None"
 
         task_node = self._conform_task_def(task_node)
         if self.butler is not None:
             self.butler.registry.refresh()
 
-        # Catch any exception and make a report based on that.
-        try:
-            result = self._execute(task_node, quantum)
-            self.report = QuantumReport(dataId=quantum.dataId, taskLabel=task_node.label)
-            return result
-        except Exception as exc:
-            self.report = QuantumReport.from_exception(
-                exception=exc,
-                dataId=quantum.dataId,
-                taskLabel=task_node.label,
-            )
-            raise
```

Comment:
Now I see the problem -- the `BaseException` is bypassing this block, but the code assumes `self.report` has been initialized one way or another. The same would happen with the block(s) moved to `MpGraphExecutor`.

Since catching `BaseException` is strongly discouraged (classes that are not `Exception` are those where premature catching would be directly harmful), would it make more sense to respec `getReport` as raising if execution was not attempted or was interrupted, and have the calling code catch `RuntimeError`?

Reply:
Problem with `getReport()` was that it relied on the state (`self.report`) and that state is hard to manage consistently in the presence of random exceptions. It is much easier to get rid of that state and return everything from `execute()`.